src/features/gift/stores/giftMakerHistory.ts

```
@@ -36,109 +45,144 @@ type Actions = {
 
 type Store = State & Actions
 
-type GiftData = {
-  state: {
-    gifts: Record<DefuseUserId, GiftMakerHistory[]>
-  }
+type StorageOperation<T> = {
+  operation: () => T | Promise<T>
+  storageName: string
+  operationType: "fetch" | "set" | "update" | "remove"
 }
 
-function serializeGiftData(gift: GiftMakerHistory) {
-  return {
-    ...gift,
-    tokenDiff: Object.fromEntries(
-      Object.entries(gift.tokenDiff).map(([key, value]) => [
-        key,
-        typeof value === "bigint" ? value.toString() : value,
-      ])
-    ),
-  }
+async function executeStorageOperations<T>(
+  operations: StorageOperation<T>[]
+): Promise<(T | null)[]> {
+  return Promise.all(
+    operations.map(async ({ operation, storageName, operationType }) => {
+      try {
+        return await operation()
+      } catch (error) {
+        let action: string
+        switch (operationType) {
+          case "fetch":
+            action = "fetch from"
+            break
+          case "set":
+            action = "set data in"
+            break
+          case "update":
+            action = "update data in"
+            break
+          case "remove":
+            action = "remove data from"
+            break
+          default:
+            action = "interact with"
+        }
+        logger.error(
+          new Error(`Failed to ${action} ${storageName}`, { cause: error })
+        )
+        return null
```

You decided ignore error and then just override with new data?

---

You are right, this is important. I think it's better to do it in separate PR.